### PR TITLE
Use `Strorage` generic instead of `Context` for `Delta` and `AccessoryDelta`

### PR DESCRIPTION
**Pull request by citizen-stig**
# Description

Refactoring, that might be used for re-orgs

## Testing
Existing tests are passing

## Docs
Documentation has not been updated

### CODE REVIEW

**Review by neysofu — APPROVED** [review]
Much cleaner!

File: module-system/sov-modules-api/src/state/scratchpad.rs [hunk 1]
```diff
@@ -11,48 +11,40 @@ use crate::{Context, Spec};
...
-    witness: <<C as Spec>::Storage as Storage>::Witness,
+pub struct Delta<S: Storage> {
+    inner: S,
+    witness: S::Witness,
     cache: StorageInternalCache,
 }
 
-impl<C: Context> Delta<C> {
-    fn new(inner: <C as Spec>::Storage) -> Self {
+impl<S: Storage> Delta<S> {
```
Comment (bkolad) [inline module-system/sov-modules-api/src/state/scratchpad.rs:20]: This was changed recently by @vlopes11 from` Storage` to `Context`.

**Comment by citizen-stig** [comment]
Requesting review from @vlopes11 , as there was this change: https://github.com/Sovereign-Labs/sovereign-sdk/pull/832

@vlopes11 , can you please provide more details, if this PR will break something needed for Gas estimation?

**Review by vlopes11 — APPROVED** [review]
@citizen-stig nothing will break! The only reason it was changed is to keep the bounds with a higher level trait so changes can be centralized into a single point (ie spec)

**Review by preston-evans98 — APPROVED** [review]

**Pull request merged**